dualscreeninfo/android/src/main/kotlin/com/microsoft/reactnativedualscreen/dualscreen/DualScreenInfo.kt

```diff
@@ -0,0 +1,138 @@
+package com.microsoft.reactnativedualscreen.dualscreen
+
+import android.content.Context
+import android.graphics.Rect
+import android.view.View
+import android.view.WindowManager
+import com.facebook.react.bridge.*
+import com.facebook.react.bridge.Arguments.createMap
+import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter
+import com.microsoft.device.display.DisplayMask
+
+const val HINGE_WIDTH_KEY = "hingeWidth"
+const val IS_DUALSCREEN_DEVICE_KEY = "isDualScreenDevice"
+const val FEATURE_NAME = "com.microsoft.device.display.displaymask"
+
+class DualScreenInfo constructor(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+    private val isDualScreenDevice: Boolean
+        get() =  reactApplicationContext.packageManager.hasSystemFeature(FEATURE_NAME)
+	private val mDisplayMask: DisplayMask
+		get() = DisplayMask.fromResourcesRectApproximation(currentActivity)
+	private val rotation: Int
+		get() {
+			val wm = currentActivity?.getSystemService(Context.WINDOW_SERVICE) as WindowManager?
+			return wm?.defaultDisplay?.rotation ?: 0
+		}
+	private val hinge: Rect
+		get() {
+			// Hinge's coordinates of its 4 edges in different mode
+			// Double Landscape Rect(0, 1350 - 1800, 1434)
+			// Double Portrait  Rect(1350, 0 - 1434, 1800)
+			return mDisplayMask.getBoundingRectsForRotation(rotation)[0]
+		}
+
+	override fun getName() = "DualScreenInfo"
+
+	override fun initialize() {
+		super.initialize()
+		createSubscription()
+	}
+
+	override fun getConstants(): Map<String, Any>? {
+        val constants: MutableMap<String, Any> = HashMap()
+    	constants[HINGE_WIDTH_KEY] = 34
+		constants[IS_DUALSCREEN_DEVICE_KEY] = isDualScreenDevice
+
+    	return constants
+    }
+
+	/**
+	 * Resolving a promise to detect if device is in Spanned mode
+	 */
+    @ReactMethod
+	fun isSpanned(promise: Promise) {
+        val windowRect: Rect? = getWindowRect()
+
+		// The windowRect doesn't intersect hinge
+		windowRect?.let {
+			if (it.width() > 0 && it.height() > 0) {
+				promise.resolve(hinge.intersect(it))
+			} else {
+				promise.resolve(false)
+			}
+		}
+
+		promise.reject(RuntimeException())
+	}
+
+	/**
+	 * Resolving a promise with window rects
```

Comment:
> window rects [](start = 28, length = 13)

Is this code returning physical pixels?  The rects should be in device-independent-pixels (DIPs).

Reply:
good point